daemon/cluster/services.go

```diff
@@ -134,6 +138,35 @@ func (c *Cluster) CreateService(s types.ServiceSpec, encodedAuth string, queryRe
 
 		switch serviceSpec.Task.Runtime.(type) {
 		// handle other runtimes here
+		case *swarmapi.TaskSpec_Generic:
```

Comment:
It looks like `UpdateService` errors out if the runtime is not "container":

```go
                newCtnr := serviceSpec.Task.GetContainer()
                if newCtnr == nil {
                        return errors.New("service does not use container tasks")
                }
```

Doesn't that method need to be updated too?